packages/extension-webgazer/src/index.ts

```diff
@@ -0,0 +1,296 @@
+import { JsPsych, JsPsychExtension, JsPsychExtensionParameters } from "jspsych";
+
+// we have to add webgazer to the global window object because webgazer attaches itself to
+// the window when it loads
+declare global {
+  interface Window { webgazer: any; }
+}
+
+interface InitializeParameters extends JsPsychExtensionParameters {
+  round_predictions: boolean;
+  auto_initialize: boolean;
+  sampling_interval: number;
+  webgazer: any;
+}
+
+interface OnStartParameters extends JsPsychExtensionParameters {
+  targets: Array<string>;
+}
+
+class WebGazerExtension implements JsPsychExtension {
+
+  constructor(private jsPsych: JsPsych) { }
+
+  // private state for the extension
+  // extension authors can define public functions to interact
+  // with the state. recommend not exposing state directly
+  // so that state manipulations are checked.
+  private currentTrialData = [];
+  private currentTrialTargets = {};
+  private currentTrialSelectors;
+  private domObserver;
+  private webgazer;
+  private initialized = false;
+  private currentTrialStart;
+  private activeTrial = false;
+  private sampling_interval;
+  private round_predictions;
+  private gazeInterval;
+  private gazeUpdateCallbacks;
+  private currentGaze;
+
+  // required, will be called at jsPsych.init
+  // should return a Promise
+  initialize(params: InitializeParameters): Promise<void> {
+    // setting default values for params if not defined
+    params.round_predictions = typeof params.round_predictions === 'undefined' ? true : params.round_predictions;
+    params.auto_initialize = typeof params.auto_initialize === 'undefined' ? false : params.auto_initialize;
+    params.sampling_interval = typeof params.sampling_interval === 'undefined' ? 34 : params.sampling_interval;
```

Comment:
You can destructure params in place and at the same time set default values:
```suggestion
  initialize({ round_predictions = true, auto_initialize = false, sampling_interval = 34 }: InitializeParameters): Promise<void> {
```

If you'd like to keep the params object:
```suggestion
  initialize(params: InitializeParameters): Promise<void> {
    // setting default values for params if not defined
    params = {
      round_predictions: true,
      auto_initialize: false,
      sampling_interval: 34,
      ...params
    };
```
https://www.freecodecamp.org/news/es5-to-esnext-heres-every-feature-added-to-javascript-since-2015-d0c255e13c6e/#default-parameters